spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/route/builder/GatewayFilterSpec.java

```diff
@@ -205,6 +208,23 @@ public GatewayFilterSpec addResponseHeader(String headerName, String headerValue
 				.apply(c -> c.setName(headerName).setValue(headerValue)));
 	}
 
+	/**
+	 * A filter that adds a local cache for storing response body for repeated requests.
+	 * <p>
+	 * If `timeToLive` and `size` are null, a global cache is used configured by the global configuration
+	 * {@link org.springframework.cloud.gateway.filter.factory.cache.LocalResponseCacheProperties}.
+	 *
+	 * @param timeToLive time an entry is kept in cache.
+	 *                   Default: 5 minutes
+	 * @param size       size expression to limit cache size (See format in {@link DataSize}.
+	 *                   Default: {@code null} (no limit)
+	 * @return a {@link GatewayFilterSpec} that can be used to apply additional filters
+	 */
+	public GatewayFilterSpec localResponseCache(@Nullable Duration timeToLive, @Nullable String size) {
```

Comment:
Add an overridden version that takes a `DataSize` rather than String. The config class should have a `DataSize` type, not String.